### PR TITLE
Read elm-analyse.json

### PR DESCRIPTION
**Pull request by jmbockhorst**

### CODE REVIEW

File: src/providers/diagnostics/elmLsDiagnostics.ts [hunk 1]
```diff
@@ -330,32 +359,86 @@ export class ElmLsDiagnostics {
...
+    }
+
+    let elmAnalyseJson = {};
+    try {
+      elmAnalyseJson = require(path.join(
+        workspacePath,
+        "elm-analyse.json",
+      )) as IElmAnalyseJson;
+    } catch {
+      // Not found
```
Comment (Razzeee) [inline src/providers/diagnostics/elmLsDiagnostics.ts:435]: We should probably log?

File: src/providers/diagnostics/elmLsDiagnostics.ts [hunk 2]
```diff
@@ -330,32 +359,86 @@ export class ElmLsDiagnostics {
...
+          : this.getFullyAppliedOperatorAsPrefixDiagnostics(tree)),
+        ...(elmAnalyseJson.checks?.UnusedTypeAlias === false
+          ? []
+          : this.getUnusedTypeAliasDiagnostics(tree)),
+        ...(elmAnalyseJson.checks?.UnusedValueConstructor === false
+          ? []
+          : this.getUnusedValueConstructorDiagnostics(tree)),
       ];
     } catch (e) {
       console.log(e);
```
Comment (Razzeee) [inline src/providers/diagnostics/elmLsDiagnostics.ts:416]: Can you log this via connection too?

File: src/providers/diagnostics/elmLsDiagnostics.ts [hunk 3]
```diff
@@ -330,32 +359,86 @@ export class ElmLsDiagnostics {
     treeContainer: ITreeContainer,
     elmWorkspace: IElmWorkspace,
   ): IDiagnostic[] => {
+    const elmAnalyseJson = this.getElmAnalyseJson(
```
Comment (Razzeee) [inline src/providers/diagnostics/elmLsDiagnostics.ts:362]: I'm wondering if this needs to do a findup from the folder where the file is that we're analyzing.

Comment (jmbockhorst) [reply src/providers/diagnostics/elmLsDiagnostics.ts:364]: Maybe, but I'm not sure how often that case would occur. For performance reasons I didn't want to be checking for it every time.

Comment (Razzeee) [reply src/providers/diagnostics/elmLsDiagnostics.ts:364]: I guess we're good, https://stil4m.github.io/elm-analyse/#/configuration seems like it's not the case.
> By default all checks are enabled. To disable checks, add an elm-analyse.json file to the root of the Elm project (beside elm.json).

**Pull request merged**